lib/authenforce/enforcers.go

```diff
@@ -1,31 +1,68 @@
 package authenforce
 
 import (
+	"context"
+	"sync/atomic"
+
 	"github.com/cuvva/cuvva-public-go/lib/cher"
-	"github.com/wearemojo/mojo-public-go/lib/authparsing"
+	"github.com/wearemojo/mojo-public-go/lib/errgroup"
+	"github.com/wearemojo/mojo-public-go/lib/merr"
+)
+
+type (
+	Enforcers   []Enforcer
+	Enforcer    func(context.Context, any, map[string]any) (bool, error)
+	SubEnforcer func(context.Context, any, map[string]any) error
 )
 
-type Enforcer func(*authparsing.AuthState) error
+const ErrNotHandled = merr.Code("auth_not_handled")
 
-func UnsafeNoAuthentication(*authparsing.AuthState) error {
-	return nil
+func UnsafeNoAuthentication(_ context.Context, _ any, _ map[string]any) (bool, error) {
+	return true, nil
 }
 
-func AllowAny(state *authparsing.AuthState) error {
+func AllowAny(_ context.Context, state any, _ map[string]any) (bool, error) {
 	if state == nil {
-		return cher.New(cher.Unauthorized, nil)
+		return true, cher.New(cher.Unauthorized, nil)
```

Comment:
possibly hoist, but then you'll need to decide the code later based on whether auth is nil